gateway-service/src/main/java/org/zowe/apiml/gateway/security/service/schema/ZosmfScheme.java

```diff
@@ -96,21 +104,25 @@ public void apply(InstanceInfo instanceInfo) {
 
             Optional<AuthSource> authSourceOptional = authSourceService.getAuthSourceFromRequest();
             authSourceOptional.ifPresent(authSource -> {
+
                 // parse authentication source to detect the source (z/OSMF / Zowe)
                 AuthSource.Parsed parsedAuthSource = authSourceService.parse(authSource);
+                // translate to JWT auth source
+                if (AuthSource.Origin.X509.equals(parsedAuthSource.getOrigin())) {
```

Comment:
At a first glance, this "if" can be substituted with extra case like that:
`case ZOSMF:
case Z509:`